### PR TITLE
Added definition of WSAOVERLAPPED

### PR DESCRIPTION
**Pull request by marler8997**

### CODE REVIEW

File: src/core/sys/windows/winsock2.d [hunk 1]
```diff
@@ -740,7 +740,21 @@ struct hostent
 }
 
 // Note: These are Winsock2!!
-struct WSAOVERLAPPED;
+struct WSAOVERLAPPED
+{
+    ulong Internal;
```
Comment (ZombineDev) [inline src/core/sys/windows/winsock2.d]: In the Windows API these are defined as `ULONG_PTR`, and so D analog is either `size_t`, or `core.stdc.stdint.uintptr_t` if you want to be pedantic about the use of ISO C terminology.

**Comment by ZombineDev** [comment]
BTW, in the future make sure to post a link to the definition of function/type you're adding in the commit message and PR description. In this case:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms741665(v=vs.85).aspx

File: src/core/sys/windows/winsock2.d (hunk 1 above)
Comment (ZombineDev) [reply src/core/sys/windows/winsock2.d]: Or perhaps better, use the Windows typedef types directly - that way there should be no room for error, or at least only one place to fix.

**Comment by rainers** [comment]
WSAOVERLAPPED should just alias OVERLAPPED, as this is what's done in the SDK/MinGW headers, too.

**Pull request closed without merging**